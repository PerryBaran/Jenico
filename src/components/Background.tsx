import React from 'react';


const Background = (props: { src: string; }) => {
    const {src} = props

    return (
        <div className='background'>
            <video 
                autoPlay 
                loop 
                muted
                src={src}
            />
      </div>
    );
};


export default Background;