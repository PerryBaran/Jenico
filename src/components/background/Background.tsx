import React from 'react';
import style from './background.module.css';


const Background = (props: { src: string; }) => {
    const {src} = props

    return (
        <div className={style.container}>
            <video 
                autoPlay 
                loop 
                muted
                src={src}
            />
      </div>
    );
};


export default Background;