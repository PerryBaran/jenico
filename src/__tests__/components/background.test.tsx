import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Background from "../../components/background/Background";

describe("Background", () => {
  const validProps = {
    src: "string",
    video: true,
  };

  afterEach(cleanup);

  describe("image", () => {
    test("image snapshot", () => {
      const { asFragment } = render(<Background src={validProps.src} />);

      expect(asFragment()).toMatchSnapshot();
    });

    test("image props", () => {
      render(<Background src={validProps.src} />);
      const image = screen.getByRole("img");

      expect(image).toHaveAttribute("src", validProps.src);
      expect(image).toHaveAttribute("alt", "background");
    });
  });

  describe("video", () => {
    test("video snapshot", () => {
      const { asFragment } = render(
        <Background src={validProps.src} video={validProps.video} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    test("video props", () => {
      render(<Background src={validProps.src} video={validProps.video} />);
      const video = screen.getByTestId("video");

      expect(video).toHaveAttribute("src", validProps.src);
      expect(video).toHaveAttribute("autoPlay");
      expect(video).toHaveAttribute("loop");
      // no way to test muted attribute
    });
  });
});
