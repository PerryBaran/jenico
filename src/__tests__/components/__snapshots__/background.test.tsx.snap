// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background image image snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="background"
      src="string"
    />
  </div>
</DocumentFragment>
`;

exports[`Background video video snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <video
      autoplay=""
      class="hide"
      data-testid="video"
      loop=""
      src="string"
    />
  </div>
</DocumentFragment>
`;
